python/src/dataset.rs

```diff
@@ -53,6 +54,91 @@ const DEFAULT_NPROBS: usize = 1;
 const DEFAULT_INDEX_CACHE_SIZE: usize = 256;
 const DEFAULT_METADATA_CACHE_SIZE: usize = 256;
 
+#[pyclass(name = "_Operation", module = "_lib")]
+#[derive(Clone)]
+pub struct Operation {
+    inner: LanceOperation,
+}
+
+fn convert_fragments(fragments: Vec<&PyAny>) -> PyResult<Vec<Fragment>> {
+    fragments
+        .iter()
+        .map(|f| f.extract::<FragmentMetadata>().map(|fm| fm.inner))
+        .collect::<PyResult<Vec<Fragment>>>()
+}
+
+fn convert_schema(schema: &PyAny) -> PyResult<Schema> {
+    let arrow_schema = ArrowSchema::from_pyarrow(schema)?;
+    Schema::try_from(&arrow_schema).map_err(|e| {
+        PyValueError::new_err(format!(
+            "Failed to convert Arrow schema to Lance schema: {}",
+            e
+        ))
+    })
+}
+
+#[pymethods]
+impl Operation {
+    fn __repr__(&self) -> String {
+        "".to_string()
```

Comment:
Perhaps:
```suggestion
        format!("{}", self.0)
```

Reply:
I had to use `{:?}` since the operations don't implement display.  This might be overly verbose for a python __repr__ but this is all internal anyways so I don't expect it to be used for much beyond debugging.